### ISSUE TITLE
Feature request: Moving and parked vehicles

### ISSUE THREAD

**Issue by DynV** [RigsOfRods/rigs-of-rods#3036]
Up to now I only did single play and I think maps lack "life". It would be nice if there were other vehicles. For performance issues I think there should be levels of "population", maybe they can be based on how much road there is on the map, also the vehicles can be rated for level of details; ie at minimum "occupancy" only the quickest to render vehicles are included (beside the one the player is in) per 1/4 mile moving at avg 15 mph.

Maybe a 2nd phase can be for the roads to be better analyzed and longer stretches have a higher avg speed, and sections with many intersections have a lower avg speed. Maybe a 3rd phase could ensure the vast majority of the time the 2 road types are proportionally "populated", ie for a maps with 1/3 rural/highway roads, there's almost always ~1/3 of the "occupying" vehicles on one of those (rural/highway road).

Thank you for your consideration

**Comment by only-a-ptr** [comment]
Hello @DynV,
Maps could indeed use more life and there are projects and tools to make it happen:
- the mission system: https://github.com/RigsOfRods/rigs-of-rods/pull/3018
 - the scripting system https://forum.rigsofrods.org/threads/angelscript-scripting-quickstart-guide.3720/

The map format already supports spawning parked vehicles. We have an AI and waypointing system, all editable and invocable from script. See demo_script.as for the AI controls and road_editor.as for road/waypoint editing. Finally, If you wanted to make NPC people to walk around, there is this: https://github.com/RigsOfRods/rigs-of-rods/pull/3030

Have fun and come chat on Discord.

**Comment by Miner34dev** [comment]
Extra idea: What if only the nearest vehicles actually had full physics? Vehicles far from the player could be kinda "dummy vehicles" with a simplistic N/B and lower LOD meshes. There are two problems with this though:

- Loading vehicles causes a bit of stuttering at the moment, so such a LOD thing would make the game unplayable currently.
- Reloading vehicles would be pretty noticeable as you would see a car respawning, which is definitely not optimal, Especially because the full N/B would be used only from very close.

**Comment by ohlidalp** [comment]
All good ideas and all _should_ be doable via scripting, though it isn't at the moment. I have 2 solutions in mind:

## The full-on fake vehicles (no collisions) approach

If you want to get your hands dirty, I suggest you download a github build of https://github.com/RigsOfRods/rigs-of-rods/pull/3030, then load the .truck file manually [as shown in this example script](https://github.com/RigsOfRods/rigs-of-rods/blob/master/resources/scripts/example_RigEditor_alpha.as#L257-L268) and then load it's meshes manually as shown in[ this other example script](https://github.com/RigsOfRods/rigs-of-rods/pull/3030/files#diff-ed909ee2ae8f45079fed6f7234433f9f083726b50d8c4e7f0265e35158bfb4d5R85-R103). Yes this approach is messy and fake but also powerful. You could in fact use more of the 'characterPosing' example to add pedestrians or other NPCs to terrain.

## The 'physics freezing' approach

The game already has an option to freeze physics for individual vehicles (Ctrl+J I think). AFAIK it cannot be toggled from script yet, but adding that is minimal work. Once that is in place, all that is left is a function to move/rotate the frozen vehicle. Also not hard to add. I think I'll go this way about it.

**Comment by ohlidalp** [comment]
There is now a concrete plan how to implement this: https://github.com/RigsOfRods/rigs-of-rods/issues/3441
